lib/manageiq/automation_engine/logger.rb

```diff
@@ -0,0 +1,52 @@
+module ManageIQ
+  module AutomationEngine
+    class Logger < ManageIQ::Loggers::Base
+      def add(severity, message = nil, progname = nil, resource_id: nil)
+        # Copied from Logger#add
+        severity ||= UNKNOWN
+        if @logdev.nil? or severity < level
+          return true
+        end
+        if progname.nil?
+          progname = @progname
+        end
+        if message.nil?
+          if block_given?
+            message = yield
+          else
+            message = progname
+            progname = @progname
+          end
+        end
```

Comment:
I wish that was extracted in upstream Ruby - I'm considering pull this out into a private method upstream.

Reply:
Actually, what they're probably going to suggest is to create a different LogDevice.  I might consider that.